src/go/rpk/pkg/api/admin/api_topic_recovery.go

```diff
@@ -0,0 +1,98 @@
+// Copyright 2023 Redpanda Data, Inc.
```

Comment:
Can we name the files api_cloud_storage.go ? so it's not tied only to topic recovery; we can add future calls to the cloud_storage API resource later in this file.